File: problems/977/solution.py
class Solution:
    def sortedSquares(self, A: List[int]) -> List[int]:
        l, r, pos, res = 0, len(A) - 1, len(A) - 1, [0] * len(A)
        
        while l <= r:
            if A[r] > abs(A[l]):
                res[pos] = A[r]**2
                r -= 1
            else:
                res[pos] = A[l]**2
                l += 1
            
            pos -= 1
        
        return res

File: problems/977/solution.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> ans(nums.size());
        int i = nums.size() - 1, lo = 0, hi = nums.size() - 1;
        
        while (lo <= hi) {
            if (abs(nums[lo]) > abs(nums[hi])) {
                ans[i--] = pow(nums[lo++], 2);
            } else {
                ans[i--] = pow(nums[hi--], 2);
            }
        }
        
        return ans;
    }
};